server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -649,6 +650,7 @@ public void testDispatchCompatibleHandler() {
         restController.registerHandler(RestRequest.Method.GET, "/foo", new RestHandler() {
             @Override
             public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+                assertThat(request.contentParser().useCompatibility(), is(true));
```

Comment:
Can you add an assert (or test) similar to this, but that should default to false ?

Reply:
will do. I will add assertion to the existing tests using application/json media types (non vnd) 
and will add a testcase for request application/vnd.elasticsearch+json;compatible-with=8 - asserting not using compatibility.